gengokit/httptransport/httptransport.go

```diff
@@ -101,6 +111,24 @@ func NewBinding(i int, meth *deftree.ServiceMethod) *Binding {
 		nField.LowCamelName = LowCamelName(nField.Name)
 
 		nBinding.Fields = append(nBinding.Fields, &nField)
+
+		// Emit warnings for certain cases
+		if nField.IsBaseType == false {
+			wrn := []string{
+				"%s.%s is a custom type '%s', only base types and repeated base ",
+				"types are supported. As a result, the generated HTTP ",
```

Comment:
Wait won't maps and nested types work if there is json in the body of an http request? It will just be marshaled to the type.